File: backend/app/api/routes/market.py
"""Market data endpoints."""
from fastapi import APIRouter, HTTPException, Query
from typing import List
from app.models.schemas import QuoteResponse
from app.services.market_data import get_market_data_service
from app.core.exceptions import MarketDataError

router = APIRouter(prefix="/market", tags=["market"])


@router.get("/quote/{ticker}", response_model=QuoteResponse)
def quote(ticker: str):
    try:
        return get_market_data_service().get_quote(ticker)
    except MarketDataError as e:
        raise HTTPException(status_code=502, detail=str(e))


@router.get("/history/{ticker}")
def history(ticker: str, period: str = Query("1mo"), interval: str = Query("1d")):
    try:
        return {"ticker": ticker.upper(), "data": get_market_data_service().get_history(ticker, period, interval)}
    except MarketDataError as e:
        raise HTTPException(status_code=502, detail=str(e))


@router.get("/search")
def search(q: str = Query(..., min_length=1)):
    return {"results": get_market_data_service().search_symbol(q)}

File: frontend/src/api/market.ts
import { api } from './client'
import type { Quote, HistoryPoint } from '@/types'

export async function getQuote(ticker: string): Promise<Quote> {
  const { data } = await api.get<Quote>(`/market/quote/${ticker.toUpperCase()}`)
  return data
}

export async function getHistory(
  ticker: string,
  period = '3mo',
  interval = '1d'
): Promise<{ ticker: string; data: HistoryPoint[] }> {
  const { data } = await api.get(`/market/history/${ticker.toUpperCase()}`, {
    params: { period, interval },
  })
  return data
}

export async function searchSymbol(
  q: string
): Promise<{ results: { symbol: string; name: string }[] }> {
  const { data } = await api.get('/market/search', { params: { q } })
  return data
}